app/models/miq_expression.rb

```diff
@@ -574,8 +574,8 @@ def self._to_ruby(exp, context_type, tz)
 
   def to_sql(tz = nil)
     tz ||= "UTC"
-    @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
+    sqlable_expression_data, attrs = preprocess_for_sql(@exp.deep_clone)
+    sql = Component.build(sqlable_expression_data, tz).to_sql if sqlable_expression_data.present?
```

Comment:
I'm fine with passing the timezone for now.  It's better to wait for the right abstraction than to prematurely move this.